### DISCUSSION TITLE
workers on all nodes

### DISCUSSION THREAD

**Discussion by maxcerny** [techno-tim/k3s-ansible#288] [Questions] [answered]
Simple question. I have created 5 node clusters that have 3 master 2 worker nodes no problems, but is there a way to create a cluster that has workers on all nodes? Is it as easy as adding the nodes in both master and worker?

**Comment by timothystewart6** [comment, marked as answer]
This should be as simple as removing this line right here in your `all.yaml` (line 36)

```yaml
  {{ '--node-taint node-role.kubernetes.io/master=true:NoSchedule' if k3s_master_taint else '' }}
```

This applies a taint to a node so that it is marked as a `master` and no workload will run unless they are critical.  If you've already created your cluster, you can remove that taint

```bash
kubectl taint nodes node1 node2 node3  node-role.kubernetes.io/master=true:NoSchedule-
```

(You'll need to updated the node names to match yours)


if you notice the `-` at the end, that means remove, so if you need to add the taint back you can run the same command without the `-`

**Reply by maxcerny** [reply]
Amazing, thanks

**Comment by mlanner** [comment]
I have a similar issue. I've got three nodes and I've put all of them in both the `[master]` and `[node]` sections, like so:

```
[master]
node01 ansible_ssh_host=10.10.1.51 
node02 ansible_ssh_host=10.10.1.52 
node03 ansible_ssh_host=10.10.1.53

[node]
node01 ansible_ssh_host=10.10.1.51 
node02 ansible_ssh_host=10.10.1.52 
node03 ansible_ssh_host=10.10.1.53 

[k3s_cluster:children]
master
node
```

I originally didn't catch the `{{ '--node-taint node-role.kubernetes.io/master=true:NoSchedule' if k3s_master_taint else '' }}` part and ran the playbook with it, tainting the nodes. This resulted in a broken installation where, towards the end, I get fatal errors:

```
TASK [k3s_agent : Enable and check K3s service]
fatal: [node02]: FAILED! => {"changed": false, "msg": "Unable to start service k3s-node: Job for k3s-node.service failed because the control process exited with error code.\nSee \"systemctl status k3s-node.service\" and \"journalctl -xeu k3s-node.service\" for details.\n"}
```

Checking on the nodes and trying to restart the `k3s-node.service` just keeps on failing. And I'm not seeing anything useful in the systemctl or journalctl output.

I then found this thread and ran the `reset.yml` playbook. After that I removed the above from `extra_server_args` and re-deployed. However, I'm still having the same issue with the k3s node service not starting.

On line 26 in `all.yml`, should I change it to read?

```
k3s_master_taint: false
```

Any thoughts on this and where I might have gone wrong?
